source/agora/consensus/EnrollmentManager.d

```diff
@@ -120,6 +127,9 @@ public class EnrollmentManager
 
         if (!results.empty)
             this.enroll_key = results.oneValue!(ubyte[]).deserializeFull!(Hash);
+
+        // load next height for preimage revelation
+        this.next_reveal_height = this.getNextRevealHeight();
```

Comment:
How come `setNextRevealHeight` is not called here?

Reply:
At first, the `next_reveal_height` is set by calling `setNextRevealHeight` in `updateEnrolledHeight` which is called in `addValidatedBlock` of the ledger, which is be reviewing in the PR #504